dali/kernels/imgproc/color_manipulation/equalize/lookup.cu

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+
+#include "dali/kernels/imgproc/color_manipulation/equalize/lookup.h"
+
+namespace dali {
+namespace kernels {
+namespace equalize {
+namespace lookup {
+
+__global__ void Lookup(const SampleDesc *sample_descs) {
+  auto sample_desc = sample_descs[blockIdx.y];
+  for (uint64_t idx = blockIdx.x * blockDim.x + threadIdx.x; idx < sample_desc.num_elements;
+       idx += blockDim.x * gridDim.x) {
+    const uint8_t *in = sample_desc.in;
+    uint8_t *out = sample_desc.out;
+    uint64_t channel_idx = idx % sample_desc.num_channels;
```

Comment:
why relying on modulus to index the channels, when you could use the y dimenson (threadIdx.y)?